### PR TITLE
Adding DaemonSet support

### PR DESCRIPTION
**Pull request by eranreshef**

### CODE REVIEW

File: kubetest/client.py [hunk 1]
```diff
@@ -110,6 +110,27 @@ def load_statefulset(self, path, set_namespace=True):
...
+
+        Args:
+            path (str): The path to the DaemonSet manifest.
+            set_namespace (bool): Enable/disable the automatic
+                augmentation of the DaemonSet namespace.
+
+        Returns:
+            objects.DaemonSet: The DaemonSet for the specified manifest.
+        """
+        log.info('loading statefulset from path: %s', path)
```
Comment (edaniszewski) [inline kubetest/client.py]:
```suggestion
        log.info('loading daemonset from path: %s', path)
```

File: kubetest/objects/daemonset.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
+"""Kubetest wrapper for the Kubernetes ``StatefulSet`` API Object."""
```
Comment (edaniszewski) [inline kubetest/objects/daemonset.py]:
```suggestion
"""Kubetest wrapper for the Kubernetes ``DaemonSet`` API Object."""
```

File: kubetest/objects/daemonset.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                Otherwise, the namespace will need to be provided.
+        """
+        if namespace is None:
+            namespace = self.namespace
+
+        log.info('creating daemonset "%s" in namespace "%s"', self.name,
+                 self.namespace)
+        log.debug('daemonset: %s', self.obj)
+
+        self.obj = self.api_client.create_namespaced_stateful_set(
```
Comment (edaniszewski) [inline kubetest/objects/daemonset.py]:
```suggestion
        self.obj = self.api_client.create_namespaced_daemon_set(
```

File: kubetest/objects/daemonset.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            client.V1Status: The status of the delete operation.
+        """
+        if options is None:
+            options = client.V1DeleteOptions()
+
+        log.info('deleting daemonset "%s"', self.name)
+        log.debug('delete options: %s', options)
+        log.debug('daemonset: %s', self.obj)
+
+        return self.api_client.delete_namespaced_daemonset(
```
Comment (edaniszewski) [inline kubetest/objects/daemonset.py]:
```suggestion
        return self.api_client.delete_namespaced_daemon_set(
```

File: kubetest/objects/daemonset.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+            name=self.name,
+            namespace=self.namespace,
+            body=options,
+        )
+
+    def refresh(self):
+        """Refresh the underlying Kubernetes DaemonSet resource."""
+        self.obj = self.api_client.read_namespaced_daemonset_status(
```
Comment (edaniszewski) [inline kubetest/objects/daemonset.py]:
```suggestion
        self.obj = self.api_client.read_namespaced_daemon_set_status(
```

**Comment by eranreshef** [comment]
apologies, don't know how I missed those...

**Comment by edaniszewski** [comment]
no worries! thanks for adding daemonset support (:

I'll likely cut a new release with all these updates later today

**Comment by eranreshef** [comment]
@edaniszewski great! looking forward to it.

**Pull request merged**